File: java/integracion_con_base_de_datos/buscar_contacto_por_nombre/src/buscar_contacto_por_nombre/buscar_contacto_por_nombre.java
package buscar_contacto_por_nombre;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.util.Scanner;
public class buscar_contacto_por_nombre {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Connection conn = conexion.conectar();
        if (conn == null) {
            System.out.println("❌ No se pudo conectar");
            sc.close();
            return;
        }
        int opcion;
        do {
            System.out.println("\n=== MENÚ ===");
            System.out.println("1. Crear contacto");
            System.out.println("2. Buscar contacto");
            System.out.println("3. Salir");
            System.out.print("Seleccione una opción: ");
            opcion = sc.nextInt();
            sc.nextLine(); 
            switch (opcion) {
                case 1:
                    System.out.println("\n=== CREAR CONTACTO ===");
                    System.out.print("Nombre: ");
                    String nombre = sc.nextLine();
                    System.out.print("Teléfono: ");
                    String telefono = sc.nextLine();
                    System.out.print("Correo: ");
                    String correo = sc.nextLine();
                    String insertSQL = "INSERT INTO contactos (nombre, telefono, correo) VALUES (?, ?, ?)";
                    try {
                        PreparedStatement ps = conn.prepareStatement(insertSQL);
                        ps.setString(1, nombre);
                        ps.setString(2, telefono);
                        ps.setString(3, correo);
                        ps.executeUpdate();
                        System.out.println("✅ Contacto guardado");
                    } catch (Exception e) {
                        System.out.println("❌ Error al guardar: " + e.getMessage());
                    }
                    break;
                case 2:
                    System.out.println("\n=== BUSCAR CONTACTO ===");
                    System.out.print("Ingrese el nombre a buscar: ");
                    String nombreBuscado = sc.nextLine();
                    String sql = "SELECT * FROM contactos WHERE nombre LIKE ?";
                    try {
                        PreparedStatement ps = conn.prepareStatement(sql);
                        ps.setString(1, "%" + nombreBuscado + "%");
                        ResultSet rs = ps.executeQuery();
                        boolean encontrado = false;
                        while (rs.next()) {
                            encontrado = true;
                            int id = rs.getInt("id");
                            String nom = rs.getString("nombre");
                            String tel = rs.getString("telefono");
                            String cor = rs.getString("correo");
                            System.out.println("------------------------");
                            System.out.println("ID: " + id);
                            System.out.println("Nombre: " + nom);
                            System.out.println("Teléfono: " + tel);
                            System.out.println("Correo: " + cor);
                        }
                        if (!encontrado) {
                            System.out.println("❌ No se encontraron contactos");
                        }
                    } catch (Exception e) {
                        System.out.println("❌ Error al buscar: " + e.getMessage());
                    }
                    break;
                case 3:
                    System.out.println("👋 Saliendo del sistema...");
                    break;
                default:
                    System.out.println("⚠️ Opción inválida");
            }
        } while (opcion != 3);
        sc.close();
    }
}

File: python/buscar_contacto_por_nombre/buscar_contacto_por_nombre.py
from conexion import conectar
def guardar_contacto(nombre, telefono, email):
    conexion = conectar()
    if conexion:
        cursor = conexion.cursor()
        sql = "INSERT INTO contactos (nombre, telefono, email) VALUES (%s, %s, %s)"
        valores = (nombre, telefono, email)
        cursor.execute(sql, valores)
        conexion.commit()
        cursor.close()
        conexion.close()
        print("✅ Contacto guardado")
def buscar_contacto(nombre_buscar):
    conexion = conectar()
    if conexion:
        cursor = conexion.cursor()
        sql = "SELECT * FROM contactos WHERE nombre LIKE %s"
        valor = (f"%{nombre_buscar}%",)
        cursor.execute(sql, valor)
        resultados = cursor.fetchall()
        if resultados:
            print("\n🔍 Resultados encontrados:\n")
            for contacto in resultados:
                print(f"ID: {contacto[0]}")
                print(f"Nombre: {contacto[1]}")
                print(f"Teléfono: {contacto[2]}")
                print(f"Email: {contacto[3]}")
                print("-" * 30)
        else:
            print("⚠️ No se encontraron coincidencias")
        cursor.close()
        conexion.close()
def main():
    print("=== Buscar Contacto ===")
    while True:
        print("\n1. Agregar contacto")
        print("2. Buscar por nombre")
        print("3. Salir")
        opcion = input("Elige una opción: ")
        if opcion == "1":
            nombre = input("Nombre: ")
            telefono = input("Teléfono: ")
            email = input("Email: ")
            guardar_contacto(nombre, telefono, email)
        elif opcion == "2":
            nombre_buscar = input("Ingrese el nombre a buscar: ")
            buscar_contacto(nombre_buscar)
        elif opcion == "3":
            print("👋 Saliendo")
            break
        else:
            print("❌ Opción inválida")
if __name__ == "__main__":
    main()